src/bos/common/types/components.py

```diff
@@ -0,0 +1,70 @@
+#
+# MIT License
+#
+# (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+Type annotation definitions for BOS components
+"""
+
+from typing import Required, TypedDict
+
+# Mapping from string labels to sets of node names
+type NodeSetMapping = dict[str, set[str]]
+
+type BootArtifacts = dict[str, str]
+type ComponentStatus = dict[str, str]
+
+class ComponentLastAction(TypedDict, total=False):
+    action: str
+    failed: bool
+    last_updated: str
```

Comment:
Should last_updated be a datetime.datetime object instead of a string? This comment applies to all of the last_updated attributes.

Reply:
I am pretty sure we store them as strings in the database, since datetime cannot be JSON-ified.

Reply:
Confirmed that we use a string. We set that field in [the _set_last_updated](https://github.com/Cray-HPE/bos/blob/7e74095c517002e446a48349562cff5874fc90d9/src/bos/server/controllers/v2/components.py#L604) function. It gets the timestamp value from [the get_current_timestamp function](https://github.com/Cray-HPE/bos/blob/7e74095c517002e446a48349562cff5874fc90d9/src/bos/common/utils.py#L57), which returns a string.

Reply:
I spoke with Jason and we are in agreement.